src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -32,12 +32,15 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
 
             properties.Headers = message.Headers.ToDictionary(p => p.Key, p => (object)p.Value);
 
-            if (message.Headers.ContainsKey(NServiceBus.Headers.EnclosedMessageTypes))
+            string messageTypesHeader;
+            if (message.Headers.TryGetValue(NServiceBus.Headers.EnclosedMessageTypes, out messageTypesHeader))
             {
-                properties.Type = message.Headers[NServiceBus.Headers.EnclosedMessageTypes].Split(new[]
+                var index = messageTypesHeader.IndexOf(',');
+
+                if (index > 0)
                 {
-                    ','
-                }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault();
+                    properties.Type = messageTypesHeader.Substring(0, index);
+                }
```

Comment:
This looks like a change in behaviour. Previously, if there were no comma in the header, we'd set `properties.Type` to be the entire header. Now, we're only setting it if there is at least one comma.